### ISSUE TITLE
Check if we need to make changes to support ForceFlush in batch processor

### ISSUE THREAD

**Issue by nilebox** [GoogleCloudPlatform/opentelemetry-operations-go#100]
See recent change https://github.com/open-telemetry/opentelemetry-go/pull/1166 (issue: https://github.com/open-telemetry/opentelemetry-go/issues/1072)

**Comment by jsuereth** [comment]
TL;DR: Our Exporter makes the assumption that it will be offloading all spans into batches and exporting as a group.

OpenTelemetry Exporters *are not allowed to do this by specification* (for some reason).   This means that users of `ForceFlush` could wind up in a state where some spans are not exported by the GCT exporter.

The only resolution here would be to "force flush" on every export call.  Given this code is also used in the collector that gives me pause.   Here's a proposed solution:


1. Update the configuration for the trace exporter with an option "batch_spans" which defaults to false (going forward)
2. If "batch_spans" is false, then we call `Flush` on `bundler` after every export.
3. When we configure this exporter in the collector, we set `batch_spans` to true.

**Comment by dashpole** [comment]
I think the collector should do batching using the exporter helper.  IIRC it doesn't belong in our exporter here.

**Comment by jsuereth** [comment]
Sure, I'll look at the exporter-helper batching in the collector and if I can migrate to that quickly, I will.  However, I think the TL;DR: is that we'll have a long-term plan to remove the `bundler` from this exporter, but we may just call `Flush` for the next release to avoid churning the design.

**Comment by punya** [comment]
@jsuereth given that the upstream OT-Go release is still ~2 weeks away, could we consider spending the time to actually remove the `bundler` now instead of deferring that work?

**Comment by ymotongpoo** [comment]
At the timing of direct exporter implementation, the spec on batching wasn't fixed and I followed the design in OC exporter. I have no objection to remove `bundler`.
